src/GitHubExtension/DeveloperId/DeveloperIdProvider.cs

```diff
@@ -296,14 +301,24 @@ private DeveloperId CreateOrUpdateDeveloperIdFromOauthRequest(OAuthRequest oauth
         return newDeveloperId;
     }
 
-    private void RestoreDeveloperIds(IEnumerable<string> loginIds)
+    private void RestoreDeveloperIds(IEnumerable<string> loginIdsAndUrls)
     {
-        foreach (var loginId in loginIds)
+        // We take loginIds or Urls here because in older versions of DevHome, we used loginIds to save credentials.
+        // In newer versions, we use Urls to save credentials.
+        // So, we need to check if loginId is currently used to save credential, and if so, replace it with URL.
+        // This is a temporary fix, and we should replace this logic once we are sure that most users have updated to newer versions of DevHome.
+        foreach (var loginIdOrUrl in loginIdsAndUrls)
         {
-            var gitHubClient = new GitHubClient(new ProductHeaderValue(Constants.DEV_HOME_APPLICATION_NAME))
+            var isUrl = loginIdOrUrl.Contains('/');

```

Comment:
Nit: Can use Uri.TryCreate

Reply:
Left a comment. There could be some GHES cases where the TryCreate fails. Leaving as is.